api/v1/storagecluster_types.go

```diff
@@ -251,10 +251,12 @@ type ManageCephRBDMirror struct {
 }
 
 // MgrSpec defines the settings for the Ceph Manager
+// +structType:=atomic
 type MgrSpec struct {
-	// EnableActivePassive can be set as true to deploy 2 ceph manager pods, one active and one standby
-	// Ceph will promote the standby mgr when the active mgr goes down due to any reason
+	// EnableActivePassive can be set as true to deploy 2 ceph manager pods(one active and one standby), in any situation no matter the other settings & constraints
 	EnableActivePassive bool `json:"enableActivePassive,omitempty"`
+	// DisableActivePassive can be set as true to deploy only 1 ceph manager pod(it will be always active with no standby), in any situation no matter the other settings & constraints
+	DisableActivePassive bool `json:"disableActivePassive,omitempty"`
```

Comment:
Having two flags can create confusion. what can `EnableActivePassive` be used for both operations?

Reply:
I wanted to give the independence to choose the number in all situations.

Let's say in a single node deployment someone wants 2 mgr or in our upcoming resource profiles in a lean profile(it will have 1 mgr), Someone wants 2 mgr. In that case, they can use the EnableActivePassive flag.

Let's say in a normal situation someone wants to use just 1 mgr, in that case they use the DisableActivePassive variable.

If we want to stick to just one var EnableActivePassive we have to make it default true, which I think will create a problem when I implement the resource profile feature as it will not be possible to know if the value has been set manually by a customer or it's just the default.

Reply:
As it's an API change we need to be a little careful when adding/removing new Fields.

> I wanted to give the independence to choose the number in all situations.
> 
> Let's say in a single node deployment someone wants 2 mgr or in our upcoming resource profiles in a lean profile(it will have 1 mgr), Someone wants 2 mgr. In that case, they can use the EnableActivePassive flag.
> 

Single Node deployment is for only development/testing purposes.

Are we going to have different mgr counts for different profiles? i assume Lean profile is meant to stick to a set of resources, if they want to change something they will change the profile. 

> Let's say in a normal situation someone wants to use just 1 mgr, in that case they use the DisableActivePassive variable.
> 

They can set the EnableActivePassive to false as well?

> If we want to stick to just one var EnableActivePassive we have to make it default true, which I think will create a problem when I implement the resource profile feature as it will not be possible to know if the value has been set manually by a customer or it's just the default.

lets get to a conclusion on that before making any API change, based on the discussion and feedback we can see do we really need two fields or if we can have only one which is default set to true.

Reply:
> i assume Lean profile is meant to stick to a set of resources, if they want to change something they will change the profile.

Travis was suggesting in an earlier comment in lean profile we can have 1 MGR & in Balance profile we can have 2.

> They can set the EnableActivePassive to false as well?
Yes, But then false is the default for a bool value so how do we know if it's manually set by someone or it's the default

> lets get to a conclusion on that before making any API change, based on the discussion and feedback we can see do we really need two fields or if we can have only one which is default set to true.
Sure open to suggestions & discussion to have an optimal outcome

Reply:
> if we allow overriding the mgr count wont it cause any problems?

I also have the same question, If we will have profiles for different situations, Do we really need to provide the flexibility to override the mgr numbers? Initially my plan was to just deprecate the EnableActivepassive flag & control the mgr numbers from the profile. 1 in Lean mode, 2 in balanced & performance

Reply:
What would give the most flexibility is if the new flag is a `bool*` so that by default (if it's nil) we use our defaults based on profiles, and if the user sets it to `true` or `false` we override the defaults. But I agree this is likely more flexibility than we need. With the introduction of the profiles, we should have enough flexibility already. I can't think why the user would have an opinion about how many mgrs are running. So we could go with your original change to deprecate the old setting and not add a new setting.